src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -85,6 +89,49 @@ class ResultsNeedFullReloadError(RuntimeError):
     """
 
 
+@attr.s(slots=True, hash=False)
+class UQResultMetadata:
+    global_gsa_meta_data: Dict[str, Any] = attr.ib(
+        validator=attr.validators.instance_of(dict)
+    )
+    result_directory: Path = attr.ib(validator=attr.validators.instance_of(Path))
+
+    @classmethod
+    def empty(cls, result_directory: Path) -> Optional["UQResultMetadata"]:
+        return UQResultMetadata(
+            global_gsa_meta_data={}, result_directory=result_directory
+        )
+
+    @classmethod
+    def get_meta_data_from_file(
+        cls, result_directory: Path
+    ) -> Optional["UQResultMetadata"]:
```

Comment:
```suggestion
    def get_meta_data_from_dir(
        cls, result_directory: Path
    ) -> Optional["UQResultMetadata"]:
```

The name of the method should match the parameter I think